### PR TITLE
Added types definition for using with TypeScript

### PR DESCRIPTION
**Pull request by gebeto**

### CODE REVIEW

**Comment by nicholas1513** [comment]
I fully support this going in ASAP! 🎉

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+export interface DatButtonProps extends DatUnChangableFieldProps {
+    onClick: (e: React.MouseEvent) => any;
+}
+export class DatButton extends React.Component<DatButtonProps, any> { }
+
+
+export interface DatFolderProps extends DatUnChangableFieldProps {
+    title: string;
+    children: React.ReactNode[];
```
Comment (nicholas1513) [inline index.d.ts]: NOTE [1]
Similar issue to 'NOTE [0]'.   Switching this to 'any' allowed me to work with the repo's provided example. https://github.com/claus/react-dat-gui/blob/master/example/src/App.js

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+export class DatFolder extends React.Component<DatFolderProps, any> { }
+
+
+export interface DatPresetsProps extends DatUnChangableFieldProps {
```
Comment (nicholas1513) [inline index.d.ts]: export interface DatPresetsProps extends DatUnChangableFieldProps {
    onUpdate: (data: any) => any;
    options: {
        presetName?: any;
        data?: any; // Initial data
        preset: any; // Your preset
}
}
export class DatPresets extends React.Component<DatPresetsProps, any> {}

Comment (nicholas1513) [reply index.d.ts]: I thought that they may be required at first too, but it isn't used in the example code.

Comment (gebeto) [reply index.d.ts]: Okay, got it 👍

**Comment by rohan-deshpande** [comment]
Not a TypeScript user, so I'm just wondering - is there any benefit in having a TypeScript example or something in the docs guiding people on how to get started with using the component with TypeScript?

**Comment by gebeto** [comment]
@rohan-deshpande in the docs, types of parameters are discribed using TypeScript similar syntax, it just describe a types of parameters, so yep, it is easier to understand what type of parameter can be in each type of component

**Comment by gebeto** [comment]
any updates?

**Comment by rohan-deshpande** [comment]
On vacation at the moment will have to take a final look when I get back in a week or so.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    children: any; // The dat.GUI components that make up the controller
+    liveUpdate?: boolean;           // Determines if live updates should occur, defaults to true
+    labelWidth?: number;            // The width of the labels in pixels, defaults to 40
+    className?: string;             // The class name to set on the DatGui div
+    style?: StyleSheet;             // The style object to set on the DatGui div
+}
+
+export default class DatGui extends React.Component<DatGuiProps, any> { }
+
+export class DatUnChangableFieldProps {
```
Comment (rohan-deshpande) [inline index.d.ts]: should be `DatUnchangeableFiledProps` as "unchangeable" is a single word

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+export interface DatButtonProps extends DatUnChangableFieldProps {
+    onClick: (e: React.MouseEvent) => any;
+}
+export class DatButton extends React.Component<DatButtonProps, any> { }
+
+
+export interface DatFolderProps extends DatUnChangableFieldProps {
+    title: string;
+    children: any;
+}
```
Comment (rohan-deshpande) [inline index.d.ts]: As of #7 `DatFolder` now has a new `closed` prop, might wanna add it

File: index.d.ts (hunk 2 above)
Comment (rohan-deshpande) [reply index.d.ts]: Hmm, it should perhaps be required, it maybe doesn't make much sense to have a preset without a name.

**Comment by gebeto** [comment]
All is ok) I've done:
 - [x] add and remove extra  lines
 - [x] rename `UnChangable` to `Unchangable`
 - [x] add `closed` on `DatFolder`

**Comment by rohan-deshpande** [comment]
@gebeto wicked! Looks good to me, merging.

**Pull request merged**